hamilton/execution/executors.py

```diff
@@ -162,6 +166,36 @@ def base_execute_task(task: TaskImplementation) -> Dict[str, Any]:
     return final_retval
 
 
+@dataclasses.dataclass
+class TaskFutureWrappingFunction(TaskFuture):
+    """Wraps a python function call in a TaskFuture."""
+
+    def __init__(self, function: Callable[[], Any]):
+        self.function = function
+        self._results = None
+        self._done = False
+        self._exception = None
+
+    def get_state(self):
+        try:
+            self._results = self.function()
```

Comment:
Shouldn't this only call the function once? E.G. it should be cached?

Reply:
Yup! This was an oversight. Commit inbound.